src/System.Management.Automation/CoreCLR/CorePsAssemblyLoadContext.cs

```diff
@@ -587,4 +587,26 @@ public static void SetPowerShellAssemblyLoadContext([MarshalAs(UnmanagedType.LPW
             PowerShellAssemblyLoadContext.InitializeSingleton(basePaths);
         }
     }
+
+    /// <summary>
+    /// Provides helper functions to faciliate calling managed code from a native PowerShell host.
+    /// </summary>
+    public static unsafe class PowerShellUnsafeAssemblyLoad
+    {
+        /// <summary>
+        /// Load an assembly in memory from unmanaged code.
+        /// </summary>
+        /// <param name="data">
+        /// Unmanaged pointer to assembly data buffer.
+        /// </param>
+        /// <param name="size">
+        /// Size in bytes of the assembly data buffer.
+        /// </param>
+        [UnmanagedCallersOnly]
+        public static void LoadAssemblyFromNativeMemory(IntPtr data, int size)
+        {
+            using var stream = new UnmanagedMemoryStream((byte*)data, size);
+            AssemblyLoadContext.Default.LoadFromStream(stream);
```

Comment:
LoadFromStream will dispose the stream.
```suggestion
            AssemblyLoadContext.Default.LoadFromStream(new UnmanagedMemoryStream((byte*)data, size));
```

Reply:
And doc comment says https://docs.microsoft.com/en-us/dotnet/api/system.io.unmanagedmemorystream?view=net-5.0#remarks
>This type implements the IDisposable interface, but does not actually have any resources to dispose. This means that disposing it by directly calling Dispose() or by using a language construct such as using (in C#) or Using (in Visual Basic) is not necessary.